lib/wibox/widget/base.lua

```diff
@@ -485,6 +496,14 @@ local function drill(ids, content)
         end
     end
 
+    -- Let some containers handle the template themselves.
+    -- This can be used for use cases such as lazy loading, repeaters or
+    -- conditional loaders.
+    if l._accept_templates and l.set_templates then
```

Comment:
The historical reason why this PR was written was for the client layout system. See https://elv13.github.io/client_tiling/dynamite.html

For the last couple years, this was mentioned over and over for other reasons. You added the `only_on_screen` container, which is very useful. But the "screen exist" isn't the only use use. This is a more generic version.

It is correct that this doesn't "really really" have to use the template hook API. However that API will also be used for the `gears.connection` API. It has been in my git stash for some years. It solves the problem where it is often hard to access other objects/widgets from within a declarative construct. It's a 1:1 port of the Qt-QML equivalent (same use case, same syntax). So, since this mechanism is needed anyway, better use it and make the API 10% better than a "pure" self contained one.

In retrospect, I should probably have added the `gears.connection` part first. This would have made the `connected_global_signals` API nicer. I will try to address your comments first, then see if I can revive this zombie code and add it to the PR. If I recall, it had many unsolved problems...